eth/downloader/peer.go

```diff
@@ -441,6 +483,20 @@ func (ps *peerSet) BlockIdlePeers() ([]*peer, int) {
 	return ps.idlePeers(61, 61, idle, throughput)
 }
 
+// HeaderIdlePeers retrieves a flat list of all the currently header-idle peers
+// within the active peer set, ordered by their reputation.
+func (ps *peerSet) HeaderIdlePeers() ([]*peer, int) {
+	idle := func(p *peer) bool {
+		return atomic.LoadInt32(&p.headerIdle) == 0
+	}
+	throughput := func(p *peer) float64 {
```

Comment:
Why isn't this a method on the peer itself? It seems a bit odd to create a lambda when it's mandatory that all peers should be able return their header throughput anyway. If a peer isn't considered a *header idle peer* perhaps it should return `0`?